src/index.js

```diff
@@ -1,31 +1,31 @@
-import {plugin} from 'postcss';
-
 import transform from './lib/transform';
 
-export default plugin('postcss-calc', (opts) => {
+function pluginCreator(opts) {
   const options = Object.assign({
     precision: 5,
     preserve: false,
     warnWhenCannotResolve: false,
     mediaQueries: false,
     selectors: false
   }, opts);
-
-  return (css, result) => {
-    css.walk(node => {
-      const { type } = node;
-
-      if (type === 'decl') {
-        transform(node, "value", options, result);
-      }
-
-      if (type === 'atrule' && options.mediaQueries) {
+  return {
+    postcssPlugin: 'postcss-calc',
+    Declaration(node, {result}) {
+      transform(node, "value", options, result);
+    },
+    AtRule(node, {result}) {
```

Comment:
hm, maybe we should remove this from major release? What is use case for at-rule(s)?

Reply:
Since there is a `mediaQueries` option, I think the use case for at-rules is `calc()`  in media queries, like:
```css
@media (max-width: calc(40em * 2)) { ... }
```

Reply:
hm, it sounds reasonable, I think we should enable it by default